src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -865,12 +865,27 @@ otError RadioSpinel<InterfaceType, ProcessContextType>::ParseRadioFrame(otRadioF
                                                &aFrame.mInfo.mRxInfo.mTimestamp, &receiveError);
 
     VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
+    aBuffer += unpacked;
+    aLength -= unpacked;
 
     if (receiveError == OT_ERROR_NONE)
     {
         aFrame.mLength = static_cast<uint8_t>(size);
 
         aFrame.mInfo.mRxInfo.mAckedWithFramePending = ((flags & SPINEL_MD_FLAG_ACKED_FP) != 0);
+
+        if (mRadioCaps & OT_RADIO_CAPS_TRANSMIT_SEC)
+        {
+            aFrame.mInfo.mRxInfo.mAckedWithSecEnhAck = ((flags & SPINEL_MD_FLAG_ACKED_SEC) != 0);
+
+            unpacked = spinel_datatype_unpack(aBuffer, aLength, SPINEL_DATATYPE_UINT8_S SPINEL_DATATYPE_UINT32_S,
+                                              &aFrame.mInfo.mRxInfo.mAckKeyId, &aFrame.mInfo.mRxInfo.mAckFrameCounter);
+            VerifyOrExit(unpacked > 0, error = OT_ERROR_PARSE);
```

Comment:
Can you please help update `spinel.h` documentation for the newly added fields?

I would also suggest in the spinel frame format we put the new fields (keyid and ack frame counter) into a spinel struct. This way if more fields need to be added in future to spinel to frame format, for cases that we won't need these security related fields it can be added as an empty struct and skipped over during parsing (this would help with backward compatibility).

Reply:
Thank you Abtin for the review. Accept and resolve all comments.

For this one. Looks like in `spinel.h`, `SPINEL_PROP_LAST_STATUS` only specify the first status code for host-to-NCP case. The information passed along with this property in `TxDone` for host-to-RCP case is not specified. Do you want me to summarize it and add it in `spinel.h`?

Reply:
Thanks @brianljt. 👍 
I see! This is `LAST_STTUS` (I thought it was `STREAM_RAW`). I think it's fine.  

As a future enhancement, I think we should conside re-designing/changing the  the RCP spinel model (maybe add new commands and callback that are more related to the radio APIs). I never liked the idea of using `LAST_STATUS` for tx done callback. IIRC this was quick solution (more of a hack) from when link-raw was added quickly.